src/components/MultiSelect/MultiSelectDialog.tsx

```diff
@@ -113,121 +156,299 @@ export const MultiSelectDialog = chakra(
+                            isChecked={isChecked(id, childItem.id)}
+                            onChange={onChange}
+                            display={defaultChildCheckboxes}
+                          />
+                        ))}
+                      </>
+                    ) : (
+                      <Checkbox
+                        id={item.id}
+                        labelText={item.name}
+                        name={item.name}
+                        isChecked={isChecked(id, item.id)}
+                        onChange={onChange}
+                      />
+                    )}
+                  </React.Fragment>
+                ))}
+              </CheckboxGroup>
+              {showViewLabel()}
             </Box>
-          </FocusLock>
+          </>
+        );
+      };
+
+      const showViewLabel = () => {
+        if (itemsList.length > 0) {
+          return (
+            <Text
+              onClick={() => viewAllItems()}
+              __css={styles.viewAllButton}
+              id="text-btn"
+              marginTop="s"
+            >
+              {viewAllLabel}
+            </Text>
+          );
+        }
+      };
+
+      const itemsNotFound = () => {
+        if (itemsList.length === 0) {
+          return (
+            <Text id="text-btn" marginTop="xs" isItalic="true">
+              No options found
+            </Text>
+          );
+        }
+      };
+
+      // We need this for our "fake" button inside the main menu button.
+      function onKeyPress(e) {
+        const enterOrSpace =
+          e.key === "Enter" ||
+          e.key === " " ||
+          e.key === "Spacebar" ||
+          e.which === 13 ||
+          e.which === 32;
+
+        if (enterOrSpace) {
+          e.preventDefault();
+          onClear();
+          internalRef?.current.focus();
+        }
+      }
+      // Manage focus upon closing the MultiSelect
+      // React.useEffect(() => {
+      //   setPrevIsOpen(isDropDownOpen);
+      //   if (isDropDownOpen) {
+      //   // internalRef.current?.click();
+      //   }
+      // }, [isDropDownOpen]);
+
+      const viewAllItems = () => {
+        setViewAllLabel(viewAllLabel === "View all" ? "View less" : "View all");
+        setListHeight(listHeight === "260px" ? "" : "260px");
+        setListItemsCount(
+          listItemsCount === defaultItemsVisible
+            ? items.length
+            : defaultItemsVisible
+        );
+        if (listItemsCount === defaultItemsVisible) {
+          setItemsList(items);
+        } else {
+          displayDefaultItems();
+        }
+      };
+
+      const buttonTextLabel = () => {
+        return (
+          <Text __css={styles.accordianLabel} title={buttonText}>
+            {truncateText(buttonText as string, 20)}
+          </Text>
+        );
+      };
+
+      return (
+        <Box
+          id={id}
+          ref={mergedRefs}
+          __css={styles}
+          {...rest}
+          onKeyPress={onKeyPress}
```

Comment:
There's a problem with this. Users expect to use the space bar to check/uncheck a checkbox. With this set here, you cannot do that. Instead, the small button with the number of items selected is triggered.

Reply:
Consider moving this functionality inside of `MultiSelectMenuButton`.